src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScriptLS.ts

```diff
@@ -140,6 +184,169 @@
+  applyFixButton?.addEventListener('click', applyFix);
+
+  // different AI fix states
+  const fixLoadingIndicatorElem = document.getElementById('fix-loading-indicator') as HTMLElement;
+  const fixWrapperElem = document.getElementById('fix-wrapper') as HTMLElement;
+  const fixSectionElem = document.getElementById('fixes-section') as HTMLElement;
+  const fixErrorSectionElem = document.getElementById('fixes-error-section') as HTMLElement;
+
+  // generated AI fix diffs
+  const nextDiffElem = document.getElementById('next-diff') as HTMLElement;
+  const previousDiffElem = document.getElementById('previous-diff') as HTMLElement;
+  const diffSelectedIndexElem = document.getElementById('diff-counter') as HTMLElement;
+
+  const diffTopElem = document.getElementById('diff-top') as HTMLElement;
+  const diffElem = document.getElementById('diff') as HTMLElement;
+  const noDiffsElem = document.getElementById('info-no-diffs') as HTMLElement;
+
+  const diffNumElem = document.getElementById('diff-number') as HTMLElement;
+  const diffNum2Elem = document.getElementById('diff-number2') as HTMLElement;
+
+  let diffSelectedIndex = 0;
+
+  function nextDiff() {
+    if (!suggestion || !suggestion.diffs || diffSelectedIndex >= suggestion.diffs.length - 1) return;
+    ++diffSelectedIndex;
+    showCurrentDiff();
+  }
+
+  function previousDiff() {
+    if (!suggestion || !suggestion.diffs || diffSelectedIndex <= 0) return;
+    --diffSelectedIndex;
+    showCurrentDiff();
+  }
+
+  function showCurrentDiff() {
+    if (!suggestion?.diffs?.length || diffSelectedIndex < 0 || diffSelectedIndex >= suggestion.diffs.length) return;
+
+    toggleElement(noDiffsElem, 'hide');
+    toggleElement(diffTopElem, 'show');
+    toggleElement(diffElem, 'show');
+
+    diffNumElem.innerText = suggestion.diffs.length.toString();
+    diffNum2Elem.innerText = suggestion.diffs.length.toString();
+
+    diffSelectedIndexElem.innerText = (diffSelectedIndex + 1).toString();
+
+    const diffSuggestion = suggestion.diffs[diffSelectedIndex];
+
+    const filePath = suggestion.filePath;
+    const patch = diffSuggestion.unifiedDiffsPerFile[filePath];
+
+    // clear all elements
+    while (diffElem.firstChild) {
+      diffElem.removeChild(diffElem.firstChild);
+    }
+    diffElem.appendChild(generateDiffHtml(patch));
+  }
+
+  function generateDiffHtml(patch: string): HTMLElement {
+    const codeLines = patch.split('\n');
+
+    // the first two lines are the file names
+    codeLines.shift();
+    codeLines.shift();
+
+    const diffHtml = document.createElement('div');
+    let blockDiv: HTMLElement | null = null;
+
+    for (const line of codeLines) {
+      if (line.startsWith('@@ ')) {
+        blockDiv = document.createElement('div');
+        blockDiv.className = 'example';
+
+        if (blockDiv) {
+          diffHtml.appendChild(blockDiv);
+        }
+      } else {
+        const lineDiv = document.createElement('div');
+        lineDiv.className = 'example-line';
```

Comment:
I just got all of these scripts from `src/snyk/snykCode/views/suggestion/codeSuggestionWebviewScript.ts`, the only thing I changed was the class name so that it matches the existing one used for the example commit fixes.

A lot of these functions are related to switching tabs and rendering different elements. Some of this can be moved to `snyk-ls` but it would add a lot of complexity to the task and the review process so I chose to do that in a future ticket/PR. It would be a good task to do when we implement this in IntelliJ too.

Reply:
Just a note that we are currently working on the refactor from tabs to arrows for the Community Fixes / Fixed Code Examples